### PR TITLE
feat(react-native): enable hermes engine

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -2155,6 +2193,6 @@ SPEC CHECKSUMS:
   SocketRocket: abac6f5de4d4d62d24e11868d7a2f427e0ef940d
   Yoga: 055f92ad73f8c8600a93f0e25ac0b2344c3b07e6
 
-PODFILE CHECKSUM: 79119d2af4f01a2584749dc3926902e66572d896
+PODFILE CHECKSUM: d79a4f70331d49a1a0bbd657163b616621147a8b
 
-COCOAPODS: 1.16.1
+COCOAPODS: 1.15.2
```
Comment (saghul) [inline ios/Podfile.lock:2198]: Update your cocoapods pl

Comment (Calinteodor) [reply ios/Podfile.lock:2198]: Already did it. This is wip and will be fixed with future commits.

File: android/gradle.properties [hunk 2]
```diff
@@ -18,13 +18,15 @@ org.gradle.jvmargs=-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF
...
+android.bundle.enableUncompressedNativeLibs=false
+
 # This one fixes a weird WebRTC runtime problem on some devices.
 # https://github.com/jitsi/jitsi-meet/issues/7911#issuecomment-714323255
 android.enableDexingArtifactTransform.desugaring=false
 
-android.useAndroidX=true
 android.enableJetifier=true
-android.bundle.enableUncompressedNativeLibs=false
+hermesEnabled=true
```
Comment (saghul) [inline android/gradle.properties:28]: Does this need to be here? Do libraries read it? If so I think we have react ext section, would it work there?

File: android/gradle.properties [hunk 3]
```diff
@@ -18,13 +18,15 @@ org.gradle.jvmargs=-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 
+android.bundle.enableUncompressedNativeLibs=false
```
Comment (saghul) [inline android/gradle.properties:21]: Please don't re-order this, to keep the changes at a minimum.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java [hunk 4]
```diff
@@ -37,7 +37,6 @@
 
 import org.devio.rn.splashscreen.SplashScreenModule;
 import org.webrtc.EglBase;
-import org.webrtc.Logging;
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java:40]: Keep this import, IIRC it's used in a section which is commented out as an example, but it helps not having to add the import each time.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java [hunk 5]
```diff
@@ -156,9 +155,8 @@ public List<ViewManager> createViewManagers(ReactApplicationContext reactContext
         return packages;
     }
 
-    static JSCExecutorFactory getReactNativeJSFactory() {
-        // Keep on using JSC, the jury is out on Hermes.
-        return new JSCExecutorFactory("", "");
+    static HermesExecutorFactory getHermesExecutorFactory() {
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java:158]: Drop this function entirely now and add create the factory inline.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java [hunk 6]
```diff
@@ -240,7 +238,7 @@ static void initReactInstanceManager(Activity activity) {
                 .setCurrentActivity(activity)
                 .setBundleAssetName("index.android.bundle")
                 .setJSMainModulePath("index.android")
-                .setJavaScriptExecutorFactory(getReactNativeJSFactory())
+                .setJavaScriptExecutorFactory(getHermesExecutorFactory())
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java:241]:
```suggestion
                .setJavaScriptExecutorFactory(new HermesExecutorFactory())
```

File: ios/Podfile [hunk 7]
```diff
@@ -17,6 +17,13 @@ end
 target 'JitsiMeet' do
   project 'app/app.xcodeproj'
 
+  # React Native and its dependencies
+  #
+
+  use_react_native!(
```
Comment (saghul) [inline ios/Podfile:23]: The app doesn't need this. The only part that uses Hermes, as part of RN is the SDK.

File: ios/app/app.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -379,6 +379,7 @@
 				0BB7DA181EC9E695007AAE98 /* Adjust ATS */,
 				DE4F6D6E22005C0400DE699E /* Setup Dropbox */,
 				E9D850368D253EFA8AB3B8D1 /* [CP] Copy Pods Resources */,
+				72B9238B1AA29B269672EC1C /* [CP] Embed Pods Frameworks */,
```
Comment (saghul) [inline ios/app/app.xcodeproj/project.pbxproj:382]: I think this will go once you remove the unnecessary thing from the Podfile.

**Review by saghul — COMMENTED** [review]
Great stuff! Left some comments, PTAL. Please rebase with master too, so the new SDK CI tests run.

You can also have some stuff to delete:

- The part which pushes the JSC library to maven, in the android release-sdk.sh script

File: android/sdk/build.gradle [hunk 9]
```diff
@@ -91,6 +91,10 @@ dependencies {
     implementation project(':react-native-webrtc')
     implementation project(':react-native-webview')
 
+    if (hermesEnabled.toBoolean()) {
```
Comment (saghul) [inline android/sdk/build.gradle:94]: Drop the check and add the dependency below the react native one.

File: android/build.gradle [hunk 10]
```diff
@@ -17,16 +17,14 @@ buildscript {
...
+    buildToolsVersion = "34.0.0"
     compileSdkVersion = 34
     minSdkVersion    = 24
     targetSdkVersion = 34
     supportLibVersion = "28.0.0"
-
-    // NDK is now entirely compatible with Apple Silicon M1 and M2 Macs as of version 24.0.
-    // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
     ndkVersion = "26.1.10909125"
+    hermesEnabled = "true"
```
Comment (saghul) [inline android/build.gradle:27]: is this the right place?

Comment (Calinteodor) [reply android/build.gradle:27]: Yeah. Why?

Comment (Calinteodor) [reply android/build.gradle:27]: The log I added to check if hermes is enabled, checks out.

Comment (saghul) [reply android/build.gradle:27]: Ah yeah I see the ext.

**Review by saghul — APPROVED** [review]

**Comment by keremoge** [comment]
hermes and reat-native versions will be the same, so maybe no need to have a different hermesVersion variable

**Comment by saghul** [comment]
Certainly looks that way: https://mvnrepository.com/artifact/com.facebook.react/hermes-android

@Calinteodor can you remove the extra variable pl?

@keremoge thanks for the heads up!

**Pull request merged**